source/agora/test/Ledger.d

```diff
@@ -232,3 +232,42 @@ unittest
     node_1.putTransaction(backup_tx);
     network.expectBlock(Height(3));  // new block finally created
 }
+
+// Ensure that when creating a frozen UTXO, the refund is not frozen too
+// See https://github.com/bpfkorea/agora/issues/1440
+unittest
+{
+    TestConf conf = { txs_to_nominate: 1 };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    // Create a freezing tx with two outputs:
+    // 1) A refund of 1k
+    // 2) A very large frozen amount (60.999k)
+    Amount freezeAmount = network.blocks[0].txs[1].outputs[0].value;
+    // Must be under Amount.MinFreezeAmount so that the refund isn't frozen
+    freezeAmount.mustSub(1_000.coins);
+    auto tx = network.blocks[0].spendable
+        .map!(txb => txb
+              .draw(freezeAmount, WK.Keys.AA.address.only).sign(TxType.Freeze)
+        ).front;
+
+    assert(tx.outputs.length == 2);
+    network.clients[0].putTransaction(tx);
+
+    // Wait for the block to be created
+    network.expectBlock(Height(1));
+    const b1 = network.clients[0].getBlock(1);
+    assert(b1.txs.length == 1);
+
+    // Now spend the refund transaction
+    auto tx2 = TxBuilder(b1.txs[0], 0).sign();
```

Comment:
Isn't the refund amount second output?

Reply:
auto tx2 = TxBuilder(b1.txs[0], 1).sign();

Reply:
See the third commit in this PR.